github/Repository.py

```diff
@@ -3215,6 +3216,14 @@ def get_workflow(self, id_or_name):
         :rtype: :class:`github.Workflow.Workflow`
         """
         assert isinstance(id_or_name, int) or isinstance(id_or_name, str), id_or_name
+        if (
+            type(id_or_name).__name__ == "str"
+            and not id_or_name.isnumeric()
+            and (".yml" not in id_or_name and ".yaml" not in id_or_name)
+        ):
+            for workflow in self.get_workflows():
```

Comment:
Multiple workflows can have the same name, this finds only the first.

To be honest, I don't like this magic here. `id_or_name` is a unique identifier (the int id or the string file name): https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#get-a-workflow

Search for a workflow that matches the name is 1) something that can be easily done in user code 2) should be a different method like `search_workflow`, which I don't see much benefit in adding to the overcrowded `Repository` class.

Reply:
Thanks for your feedback @EnricoMi!

My two cents:
It is possible but very unlikely that two or more workflows will be named the same (at least, I have not encountered such case). The reason I created this code is that I was searching the workflow name and was getting a 404 error, and I could not understand why this is. The method's parameter expects an id or name (`id_or_name`), and I was intuitively providing a name. 

In my opinion,  it is an additional capability that fits perfectly into the `get_worklfow()` method.